### ISSUE TITLE
avoid_returning_widgets lint showing up for all the methods that return a widget

### ISSUE THREAD

**Issue by imsamgarg** [charlescyt/pyramid_lint#22]
### Is there an existing issue for this bug?

- [X] I have searched the [existing issues](https://github.com/charlescyt/pyramid_lint/issues).

### Bug Description

avoid_returning_widgets lint showing up for all the methods that return a widget. It should be restricted to the methods only in the StatelessWidgets and State classes. We can also hide it for methods that are overridden.

Following are the examples of the false lint warning in a InhertedWidget class and a PageTransitionBuilder class.

![Screenshot from 2024-01-16 16-03-51](https://github.com/charlescyt/pyramid_lint/assets/34547038/b53dc4f7-ba34-4db3-90ca-aeeb7ccb650f)
![Screenshot from 2024-01-16 16-57-29](https://github.com/charlescyt/pyramid_lint/assets/34547038/331d13f4-30d1-4477-83b3-e6c8623c5575)

### Steps to Reproduce

Just declare any method that returns a widget.

**Comment by charlescyt** [comment]
Thanks for the bug report.

I agree methods with `override` annotation should be ignored by the lint. But do we want to restrict the lint for the widget classes only? I do think the lint should check for top level functions.

For the `InheritedWidget` case, would it be enough if the lint ignores method with name `of`?
We could also add options for the lint and let users to configure what method names and annotations should the lint ignore.

**Comment by imsamgarg** [comment]
We can go either way, restrict the lint for only widget classes and top-level functions or handle method name cases. In the latter, we need to handle `of`, `maybeOf` method names or we can handle the static method case directly.

**Comment by charlescyt** [comment]
Hello! The lint now ignores static and overridden methods.
I have also added `ignored_method_names` options.

You could try the latest version with
```yaml
pyramid_lint:
  git:
    url: https://github.com/charlescyt/pyramid_lint.git
    ref: 023526a
```

**Comment by imsamgarg** [comment]
I think we should ignore it for extensions as well.
![Screenshot from 2024-01-26 00-13-35](https://github.com/charlescyt/pyramid_lint/assets/34547038/fc91b4a3-feb8-43c4-a8c0-0ebe1fbef875)

**Comment by charlescyt** [comment]
I have updated the lint to ignore extension methods. Feel free to reopen the issue if there are any problems.
